src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,99 @@
+        leftTalon.config_kP(0, robotConstants.controlConstants.leftShooterSettings.getKP());
+        leftTalon.config_kI(0, robotConstants.controlConstants.leftShooterSettings.getKI());
+        leftTalon.config_kD(0, robotConstants.controlConstants.leftShooterSettings.getKD());
+        leftTalon.config_kF(0, robotConstants.shooterConstants.LEFT_KF);
+        leftTalon.selectProfileSlot(0, 0);
+
+        rightTalon = new WPI_TalonSRX(robotConstants.can.RIGHT_SHOOTER_CONTROLLER);
+        rightTalon.configSelectedFeedbackSensor(TalonSRXFeedbackDevice.CTRE_MagEncoder_Relative, 0, 0);
+        rightTalon.configSelectedFeedbackCoefficient(1 / (robotConstants.shooterConstants.RIGHT_UNITS_PER_ROTATION * 600));
+        rightTalon.config_kP(0, robotConstants.controlConstants.rightShooterSettings.getKP());
+        rightTalon.config_kI(0, robotConstants.controlConstants.rightShooterSettings.getKI());
+        rightTalon.config_kD(0, robotConstants.controlConstants.rightShooterSettings.getKD());
+        rightTalon.config_kF(0, robotConstants.shooterConstants.RIGHT_KF);
+        rightTalon.selectProfileSlot(0, 0);
+    }
+
+    /**
+     * @param power The power to set the talon in open loop.  Value should be between -1.0 and 1.0.
+     */
+    @Override
+    public void move(double power) {
+        leftTalon.set(power);
+        rightTalon.set(power);
+    }
+
+    public void startPID() {
+        startPID(robotConstants.shooterConstants.DEFAULT_RPM);
+    }
+
+    /**
+     * Starts using velocity PID instead of open-loop.
+     *
+     * @param setpointVelocity The setpoint used for calculation the velocity error in RPM.
+     */
+    public void startPID(double setpointVelocity) {
+        leftTalon.set(ControlMode.Velocity, setpointVelocity);
+        rightTalon.set(ControlMode.Velocity, setpointVelocity);
+    }
+
+    /**
+     * @return the speed of the shooter in RPM.
+     */
+    public double getAverageSpeed() {
+        return (getLeftSpeed() + getRightSpeed()) / 2;
+    }
+
+    /**
+     * @return the speed of the shooter in RPM.
+     */
+    public double getLeftSpeed() {
+        return leftTalon.getSelectedSensorVelocity();
+    }
+
+    /**
+     * @return the speed of the shooter in RPM.
+     */
+    public double getRightSpeed() {
+        return rightTalon.getSelectedSensorVelocity();
+    }
+
+    /**
+     * @param RPM Revolution per minute
+     * @return RPM in meter per second
+     */
+    public static double rpmToMeterPerSecond(double RPM) {
+        return Units.rotationsPerMinuteToRadiansPerSecond(RPM) * robotConstants.shooterConstants.WHEEL_RADIUS;
+    }
+
+    /**
+     * @param meterPerSecond speed to be converted
+     * @return velocity in revolution per minute
+     */
+    public static double meterPerSecondToRPM(double meterPerSecond) {
+        return Units.radiansPerSecondToRotationsPerMinute(meterPerSecond / robotConstants.shooterConstants.WHEEL_RADIUS);
+    }
+}
```

Comment:
add encoder Reset method...?

Reply:
this is a calculation function.
What does it have with encoder?